### ISSUE TITLE
Creating a persistent room throws an error when using db_type sql

### ISSUE THREAD

**Issue by Jaykju** [processone/ejabberd#3722]
## Environment

- ejabberd version: 21.07
- Installed from: official Docker container `ejabberd/ecs:21.07`

## Configuration excerpt

```yaml
  mod_muc:
    # [...]
    db_type: sql
    ram_db_type: sql
    default_room_options:
      mam: true
      persistent: true
```

## Errors from error.log/crash.log

No errors

## Error from CLI

```
bin/ejabberdctl create_room test-room conference.localhost localhost
```
```
Unhandled exception occurred executing the command:
** exception error: no match of right hand side value {atomic,[]}
   in function  mod_muc_admin:maybe_store_room/4 (src/mod_muc_admin.erl, line 713)
   in call from ejabberd_ctl:call_command/4 (src/ejabberd_ctl.erl, line 341)
   in call from ejabberd_ctl:try_call_command/4 (src/ejabberd_ctl.erl, line 306)
   in call from ejabberd_ctl:process2/4 (src/ejabberd_ctl.erl, line 244)
   in call from ejabberd_ctl:process/2 (src/ejabberd_ctl.erl, line 222)
   in call from rpc:'-handle_call_call/6-fun-0-'/5 (rpc.erl, line 197)
```

## Bug description

When I try to create a room using the CLI or the admin API the error noted above is returned. The error is only thrown if `db_type` is set to `sql`. When I tried mnesia it worked fine. 

Interestingly enough the rooms are actually persisted to the database even though an error is thrown.

**Comment by licaon-kter** [comment]
Duplicate of https://github.com/processone/ejabberd/issues/3632 ?

**Comment by Jaykju** [comment]
Maybe, although I did look into that issue beforehand, and it does not mention any errors.

I also think that I am using the "next release" mentioned in that issue?

**Comment by badlop** [comment]
https://github.com/processone/ejabberd/issues/3632 got fixed in 21.07 with this commit: https://github.com/processone/ejabberd/commit/271a9f097d251fb76912f935f83b29bae833799b 

What you are experiencing is https://github.com/processone/ejabberd/issues/3700 which will be fixed in the next ejabberd release thanks to https://github.com/processone/ejabberd/commit/54c23a65db7491a9cc51eac3f89734c161603481

I can reproduce your crash with 21.07. When using git version, that command works correctly.

**Comment by Jaykju** [comment]
Awesome, and thanks for clearing that up.
